internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -117,8 +124,21 @@ func (r *ClusterDeploymentsReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 	if *cluster.Status == models.ClusterStatusInstalled && swag.StringValue(cluster.Kind) == models.ClusterKindCluster {
 		if !clusterDeployment.Spec.Installed {
 			err = r.updateClusterMetadata(ctx, clusterDeployment, cluster)
+			if err != nil {
+				return r.updateState(ctx, clusterDeployment, cluster, err)
+			}
+		}
+		// Delete Day1 Cluster
+		err = r.Installer.DeregisterClusterInternal(ctx, installer.DeregisterClusterParams{
+			ClusterID: *cluster.ID,
+		})
+		if err != nil {
 			return r.updateState(ctx, clusterDeployment, cluster, err)
 		}
+		if !r.isSNO(clusterDeployment) {
```

Comment:
Maybe instead of doing everything in a single reconcile you need to requeue and create it in a new reconcile? 
I'm not sure what is the right flow with controllers, @mhrivnak ?

Reply:
It's ok to do a little or a lot in a reconcile. But mostly the code needs to be ok with either. If you perform only the first few out of several actions, then the reconcile dies or errors out, it needs to be able to pick up where it left off the next time Reconcile runs.